File: src/main/java/com/mrad/ecommercebackend/product/ProductController.java
package com.mrad.ecommercebackend.product;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/product")
class ProductController {
    @Autowired
    private  ProductService service;
    @GetMapping("/all")
    public List<Product> allProducts() {
            return service.allProducts();
    }
}

File: src/main/java/com/mrad/ecommercebackend/product/ProductController.kt
package com.mrad.ecommercebackend.product

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/product")
class ProductController @Autowired constructor(private val service: ProductService) {

    @GetMapping("/all")
    fun allProducts(): List<Product> {
            return service.allProducts()
    }
}